src/node_buffer.cc

```diff
@@ -623,7 +623,7 @@ void StringWrite(const FunctionCallbackInfo<Value>& args) {
   Local<String> str = args[0]->ToString(env->isolate());
 
   if (encoding == HEX && str->Length() % 2 != 0)
-    return env->ThrowTypeError("Invalid hex string");
+    return env->ThrowTypeError("Invalid hex string length (must be even)");
```

Comment:
Perhaps, "The hex string length must be an even number" or "The hex string must have an even number of characters"